tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/BuilderUtil.scala

```diff
@@ -26,9 +26,60 @@ object BuilderUtil {
   /** Removes elem from the scope, as the scope only contains types of free variables */
   def markAsBound(elem: TlaEx): TBuilderInternalState[Unit] = State[TBuilderContext, Unit] { mi: TBuilderContext =>
     require(elem.isInstanceOf[NameEx])
-    (mi.copy(nameScope = mi.nameScope - elem.asInstanceOf[NameEx].name), ())
+    (mi.copy(freeNameScope = mi.freeNameScope - elem.asInstanceOf[NameEx].name), ())
   }
 
+  /** Performs shadowing checks for ternary operators which introduce bound variables */
```

Comment:
The comment makes it sound like the point of this is to preform shadowing checks, but iiuc, this is meant to construct a particular expression, and the comment is highlighting one particular check that happens during that construction? I think this can be clarified. If the point of the method is to perform these checks, seems like that should be reflected in the method name, or else the comment shouldn't indicate that is the main point?

Reply:
I've added a more verbose description.

Reply:
Thanks!